Source/Core/Common/BitUtils.h

```diff
@@ -199,4 +199,14 @@ inline To BitCast(const From& source) noexcept
   std::memcpy(&storage, &source, sizeof(storage));
   return reinterpret_cast<To&>(storage);
 }
+
+template <typename T>
+void SetBit(T& value, size_t bit_number, bool bit_value)
+{
+  if (bit_value)
+    value |= (1 << bit_number);
```

Comment:
```suggestion
    value |= (T{1} << bit_number);
```

This prevents undefined behavior in cases like `1 << N - 1` where N is the size of a type in bits (so 1 << 31 with a 32-bit unsigned, type etc).